src/bridgestan.cpp

```diff
@@ -327,9 +327,6 @@ stan::model::model_base& new_model(stan::io::var_context &data_context,
                                    unsigned int seed, std::ostream *msg_stream);
 
 model_rng* construct_impl(char* data_file, unsigned int seed, unsigned int chain_id) {
-  // enforce math lib thread locality for multi-threading
-  static thread_local stan::math::ChainableStack dummy;
```

Comment:
What's the difference between putting this line here and putting it in log_gradient_density_impl?

If it's better to go closer to where we call stan::math::gradient, then it probably should be added to log_density_impl, too, ya?

Reply:
I need to check that something I changed later wasn’t the “true” fix, but at first I was just trying to get the code closer to how it was in the original interface for this. I believe it might matter though, since I think the version in the constructor will have dropped out of scope by the time gradient gets called. 

If that theory is correct, you’re right about the log_density version, so something needs to change anyway

Reply:
It was answer "B": it does need to be in the function with the gradient call, so I added it to log_density_impl as well.